### PR TITLE
add exlusion component

### PR DESCRIPTION
**Pull request by kuda1992**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3947

### Change description ###
Created `AllocateRoleContainerComponent` component which will host the form group
Created `DescribeExclusionComponent` component

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/cases/case-feature.routes.ts [hunk 1]
```diff
@@ -37,6 +44,35 @@ export const ROUTES: Routes = [
...
+            {
+              path: 'exclusion',
+              component: AllocateRoleContainerComponent,
+            },
+            {
+              path: 'describe-exclusion',
+              component: AllocateRoleContainerComponent,
+            },
+            {
+              path: 'exclusion-self',
```
Comment (connorpgpmcelroy) [inline src/cases/case-feature.routes.ts:63]: I'm getting rid of this in my branch so would prefer if you did as well please

File: src/cases/components/describe-exclusion/describe-exclusion.component.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    {{description}}
+  </h1>
+  <div class="govuk-form-group" [ngClass]="{'form-group-error': !formGroup.get(controlName).valid && formGroup.get(controlName).touched}" >
+    <fieldset class="govuk-fieldset">
+    <textarea class="govuk-textarea govuk-!-width-one-half"
+              id="exclusion-description"
+              formControlName="text"
+              name="exclusion-description"
+              rows="5"
+              aria-describedby="exclusion description"
```
Comment (connorpgpmcelroy) [inline src/cases/components/describe-exclusion/describe-exclusion.component.html:13]: Think this is missing the '-'

File: src/cases/components/describe-exclusion/describe-exclusion.component.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { FormGroup } from '@angular/forms';
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'exui-describe-exclusion',
+  templateUrl: './describe-exclusion.component.html',
+  styleUrls: ['./describe-exclusion.component.scss']
```
Comment (connorpgpmcelroy) [inline src/cases/components/describe-exclusion/describe-exclusion.component.ts:7]: Do we need this if it's not being used?

File: src/cases/components/describe-exclusion/describe-exclusion.component.html (hunk 2 above)
Comment (kuda1992) [reply src/cases/components/describe-exclusion/describe-exclusion.component.html:13]: good spot

File: src/cases/case-feature.routes.ts (hunk 1 above)
Comment (kuda1992) [reply src/cases/case-feature.routes.ts:63]: cheers

Comment (udaydenduluri33) [reply src/cases/case-feature.routes.ts:63]: path: 'exclusion-self' should not be present

Comment (udaydenduluri33) [reply src/cases/case-feature.routes.ts]: Also you dnt need this path             {
              path: 'describe-exclusion',
              component: AllocateRoleContainerComponent,
            },

File: src/cases/containers/role-container/allocate-role-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'exui-allocate-role-container',
+  templateUrl: './allocate-role-container.component.html',
+  styleUrls: ['./allocate-role-container.component.scss']
```
Comment (connorpgpmcelroy) [inline src/cases/containers/role-container/allocate-role-container.component.ts:7]: Again, do we need this?

**Review by udaydenduluri33 — APPROVED** [review]
Thanks @kuda1992  for the changes

**Review by connorpgpmcelroy — APPROVED** [review]
Have checked and all comments addressed so LGTM

**Pull request merged**